src/data/astarte_provider.rs

```diff
@@ -78,8 +78,42 @@ impl AstarteSubscriber for Astarte {
         Ok(rx)
     }
 
-    async fn unsubscribe(&self, _astarte_node: &AstarteNode) -> Result<(), Error> {
-        Ok(())
+    async fn unsubscribe(&self, astarte_node: &AstarteNode) -> Result<(), AstarteMessageHubError> {
+        let interfaces_to_remove = {
```

Comment:
Since this is a complex assignment a brief comment could help understand what it does.
Could "All the interfaces in this node introspection that are not in the introspection of any other node." make sense?